src/instructlab/sdg/pipeline.py

```diff
@@ -32,6 +33,32 @@ def __init__(
         self.num_procs = 8
 
 
+# This is part of the public API.
+class BlockGenerationError(Exception):
+    """A BlockGenerationError occurs when a block generates an exception during
+    generation. It contains information about which block failed and why.
+    """
+
+    def __init__(self, block: Block, exception: Exception):
+        self.block = block
```

Comment:
nit: underscore prefix this if you want just `.block_name` and `.block_type` to be used?

Reply:
I left that one "public" thinking that some might like to be able to actually muck with the block itself when handling the exception.